server/app/models/host_node_stat.rb

```diff
@@ -6,10 +6,270 @@ class HostNodeStat
+        filesystem_total: { '$sum': '$filesystem.total' }
+      }
+    },
+    # Flatten network, get selected network interface
+    {
+      '$unwind': '$network',
+    },
+    {
+      '$match': {
+        'network.name': network_iface
+      }
+    },
+    {
+      '$group': {
+        _id: '$_id',
+        created_at: { '$first': '$created_at' },
+        cpu: { '$first': '$cpu' },
+        memory: { '$first': '$memory' },
+        network_name: { '$first': '$network.name' },
+        network_rx_bytes: { '$first': '$network.rx_bytes' },
+        network_rx_errors: { '$first': '$network.rx_errors' },
+        network_rx_dropped: { '$first': '$network.rx_dropped' },
+        network_tx_bytes: { '$first': '$network.tx_bytes' },
+        network_tx_errors: { '$first': '$network.tx_errors' },
+        filesystem_used: { '$first': '$filesystem_used' },
+        filesystem_total: { '$first': '$filesystem_total' }
+      }
+    },
+    # Aggregate for each minute
+    {
+      '$group': {
+        _id: {
+          year: { '$year': '$created_at' },
+          month: { '$month': '$created_at' },
+          day: { '$dayOfMonth': '$created_at' },
+          hour: { '$hour': '$created_at' },
+          minute: { '$minute': '$created_at' }
+        },
+        created_at: { '$first': '$created_at' },
+
+        cpu_num_cores: { '$avg': '$cpu.num_cores' },
+        cpu_percent_used: { '$avg': { '$add': ['$cpu.user', '$cpu.system'] } },
+
+        memory_used: { '$avg': '$memory.used' },
+        memory_total: { '$avg': '$memory.total' },
+
+        network_name: { '$first': '$network_name' },
+        network_rx_bytes: { '$avg': '$network_rx_bytes' },
+        network_rx_errors: { '$avg': '$network_rx_errors' },
+        network_rx_dropped: { '$avg': '$network_rx_dropped' },
+        network_tx_bytes: { '$avg': '$network_tx_bytes' },
+        network_tx_errors: { '$avg': '$network_tx_errors' },
+
+        filesystem_used: { '$avg': '$filesystem_used' },
+        filesystem_total: { '$avg': '$filesystem_total' }
+      }
+    },
+    # Sort and project back into useful format
+    {
+      '$sort': { 'created_at': 1 }
+    },
+    {
+      '$project': {
+        _id: 0,
+        timestamp: '$_id',
+        cpu: {
+          num_cores: '$cpu_num_cores',
+          percent_used: '$cpu_percent_used'
+        },
+        memory: {
+          used: '$memory_used',
+          total: '$memory_total'
+        },
+        network: {
+          name: '$network_name',
+          rx_bytes: '$network_rx_bytes',
+          rx_errors: '$network_rx_errors',
+          rx_dropped: '$network_rx_dropped',
+          tx_bytes: '$network_tx_bytes',
+          tx_errors: '$network_tx_errors'
+        },
+        filesystem: {
+          used: '$filesystem_used',
+          total: '$filesystem_total'
+        }
+      }
+    }])
+  end
+
+  def self.get_aggregate_stats_for_grid(grid_id, from_time, to_time, network_iface)
```

Comment:
Exposing a single "host node network interface name" parameter for grid-level aggregation isn't that great... different host nodes running on different platforms might use different network interface names.

It would be better to have each agent figure out what network interface to monitor?